File: pyg_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.vo.Specification;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface SpecificationService {
    List<TbSpecification> findAll();

    PageResult<TbSpecification> findPage(Integer page, Integer size);


    void save(Specification specification);

    Specification findOne(Long id);

    void delete(Long[] ids);

    List<Map> findSpecList(Long typeTemplateId);
}

File: pyg_shop_web/src/main/webapp/js/service/specificationService.js
app.service('specificationService', function ($http) {
    this.findAll = function () {
        return $http.get('../specification/findAll');
    };

    this.findPage = function (page, size) {
        return $http.get('../specification/findPage/' + page + "/" + size);
    };

    this.save = function (entity) {
        return $http.post('../specification/save', entity);
    };

    this.findOne = function (id) {
        return $http.get('../specification/findOne/' + id);
    };

    this.delete = function (selectIds) {
        return $http.get("../specification/delete/" + selectIds)
    };

    this.findSpecList = function (typeTemplateId) {

        return $http.get("../specification/findSpecList/" + typeTemplateId)
    }

});